### DISCUSSION TITLE
Why unresolved_mark is not present at Options, but top_level_mark is

### DISCUSSION THREAD

**Discussion by JSerFeng** [swc-project/swc#6162] [Q&A] [answered]
As far as I know, unresolved_mark is important if I want to add global API eg: require, but using swc::Compiler::process_js_with_custom_pass, seems I can't get unresolved_mark inside it, and every call to process_js_with_custom_pass, a `Mark::new()` call presents, but I can't get that mark right?

Now I can only clear mark and re-apply mark myself using `resolver()`

**Comment by kdy1** [comment, marked as answer]
I simply forgot adding it

**Reply by JSerFeng** [reply]
Will this field be added soon ?

**Reply by kdy1** [reply]
Feel free to send a PR
